libavcodec/vvc/filter.c

```diff
@@ -935,11 +942,21 @@ void ff_vvc_deblock_horizontal(const VVCLocalContext *lc, const int x0, const in
                 if (!all_zero_bs) {
                     src = &fc->frame->data[c_idx][(y >> vs) * fc->frame->linesize[c_idx] + ((x >> hs) << fc->ps.sps->pixel_shift)];
                     if (!c_idx) {
-                        fc->vvcdsp.lf.filter_luma[0](src, fc->frame->linesize[c_idx],
-                            beta, tc, no_p, no_q, max_len_p, max_len_q, horizontal_ctu_edge);
+                        fc->vvcdsp.lf.filter_luma[0](src, fc->frame->linesize[c_idx], beta, tc, no_p, no_q, max_len_p, max_len_q, horizontal_ctu_edge);
                     } else {
-                        fc->vvcdsp.lf.filter_chroma[0](src, fc->frame->linesize[c_idx],
+                           
+                        if(
+                            max_len_p[0] == 1 && max_len_q[0] == 1 &&
```

Comment:
maybe we can set max_len_p and max_len_q to 1 in the check asm.

Reply:
Yes, though this will eventually all go into the assembly to handle all cases.